helix-term/src/commands.rs

```diff
@@ -3401,6 +3402,30 @@ pub fn code_action(cx: &mut Context) {
     )
 }
 
+pub fn command_palette(cx: &mut Context) {
+    cx.callback = Some(Box::new(
+        move |compositor: &mut Compositor, _cx: &mut compositor::Context| {
+            let commands = MappableCommand::STATIC_COMMAND_LIST.iter().collect();
+            let picker = Picker::new(
+                true,
+                commands,
+                |command| match command {
+                    MappableCommand::Typable { name, args, doc } => {
+                        format!("{}: {}", name, doc).into()
+                    }
+                    MappableCommand::Static { name, fun, doc } => {
+                        format!("{}: {}", name, doc).into()
+                    }
+                },
+                move |_editor, command, _action| {
+                    log::info!("would execute: {}", command.name());
```

Comment:
Here, access to outer `cx` is needed and that would require `'static` lifetime. Would anyone please be able to help me figure out to to handle this properly?

Reply:
Cherry pick https://github.com/helix-editor/helix/pull/574/commits/3b8d5102acfcd2aeacb7d8c88eaaf51f5be0cc7b from the DAP branch. This will give you a `compositor::Context`, but then you can construct a `commands::Context` from it by setting the rest of the fields by hand (count: 1, register: None etc)

Reply:
wonderful, thanks a lot 🙏